File: app/src/main/java/com/example/rnhostv2_2/ReactOptions.java
package com.example.rnhostv2_2;

import com.facebook.jni.HybridData;

public class ReactOptions {

    static {
        System.loadLibrary("fbjni");
        System.loadLibrary("rnhostv2_2");
    }

    private native HybridData initHybrid();
    private final HybridData mHybridData;

    ReactOptions(HybridData hybridData) {
        mHybridData = hybridData;
    }

    public native String getIdentity();
    public native void setIdentity(String identity);

    ReactOptions() {
        mHybridData = initHybrid();
    }


}

File: app/src/main/java/com/example/rnhostv2_2/ReactOptions.kt
package com.example.rnhostv2_2

import com.facebook.jni.HybridData

class ReactOptions {
    private external fun initHybrid(): HybridData
    private val mHybridData: HybridData

    internal constructor(hybridData: HybridData) {
        mHybridData = hybridData
    }

    internal constructor() {
        mHybridData = initHybrid()
    }

    var identity: String?
        external get
        external set

    companion object {
        init {
            System.loadLibrary("fbjni")
            System.loadLibrary("rnhostv2_2")
        }
    }
}